src/main/java/address/MainApp.java

```diff
@@ -35,15 +41,29 @@ public void start(Stage primaryStage) {
     }
 
     protected void setupComponents() {
+
         config = getConfig();
         PreferencesManager.setAppTitle(config.appTitle);
-        AddressBookWrapper dataFromFile = StorageManager.getDataFromFile(
-                                                PreferencesManager.getInstance().getPersonFile());
+
+        final File initialFile = PreferencesManager.getInstance().getPersonFile();
+        AddressBookWrapper dataFromFile;
+        Exception fileLoadException = null;
+        try {
+             dataFromFile = StorageManager.loadDataFromSaveFile(initialFile);
+        } catch (JAXBException | FileContainsDuplicatesException e) {
+            fileLoadException = e;
+            dataFromFile = new AddressBookWrapper();
+        }
+
         modelManager = new ModelManager(dataFromFile);
         storageManager = new StorageManager(modelManager);
         mainController = new MainController(this, modelManager, config);
         syncManager = new SyncManager();
         syncManager.startSyncingData(config.updateInterval, config.isSimulateRandomChanges);
+
+        if (fileLoadException != null) {
+            EventManager.getInstance().post(new FileOpeningExceptionEvent(fileLoadException, initialFile));
+        }
```

Comment:
@damithc you can see the other half of the user error feedback logic here. It's separated because of the forced component load order